### ISSUE TITLE
Error: no reads found. Please check your Kraken report

### ISSUE THREAD

**Issue by snenish03** [jenniferlu717/Bracken#117]
Hello,

I am using kracken2, bracken and 16S_SILVA138_k2db to generate species and genus counts. 

I am having trouble generating the bracken reports where I keep getting this error - 
_Error: no reads found. Please check your Kraken report_

`kraken2 --db 16S_SILVA138_k2db --report 813.fastq.16S.kreport --report-zero-counts --threads 16  --use-names 813.fastq`
`bracken  -d kraken2/16S_SILVA138_k2db -i 813.fastq.16S.kreport -o 813.fastq.std.species.breport -r 50 -l S -t 10`

**Comment by ifanlyn95** [comment]
have u solved ur problem? I met the same error when utilized the silva and rdp

**Comment by ifanlyn95** [comment]
I have solved it by modifying the parameters, bracken -l G instead of -l S.
if ur input was the 16s results, it should be forced on the genus level or more low tax level.

**Comment by snenish03** [comment]
You are right, forcing to to genus level gives read counts. However, I am more interested in species counts using SILVA. Is that something I cannot do using Kraken?

**Comment by mihkelvaher** [comment]
This also happens when
```
database80mers.kmer_distrib
database.kraken
database80mers.kraken
```
files are empty after Bracken db build. Haven't found a cause yet.

**Comment by MatteoSchiavinato** [comment]
As a simplification to what @mihkelvaher has posted, you can also run the following command:

```
find -L library \( -name "*.fna" -o -name "*.fa" -o -name "*.fasta" \) -exec cat {} + > sequences_found.fasta 
kraken2 --db . --threads 48 sequences_found.fasta > database.kraken 
```
Thus splitting the generation of `database.kraken` in two steps. If you work with slurm workload managers, some of the brackets could be escaped weirdly and giving you a headache, splitting the step in two made my day.

**Comment by MAXINELSX** [comment]
But the data I put is meta, still showed this error

**Comment by emmaivansson** [comment]
I am trying to run bracken on paired end seq data of microbiome, and I am getting the 
'Error: no reads found. Please check your Kraken report'
despite the fact that the report exists,  it has reads and is not in mpa format. The database files aren't empty either. Have tried changing -l from default but didn't help. Running on cluster. Am at a loss here! Any input? Any other issues you've known to give this particular error?

>kraken2 --paired /proj/sens2021578/nobackup/emmai/test_kraken/readsForAssembly_collated_R1.fastq /proj/sens2021578/nobackup/emmai/test_kraken/readsForAssembly_collated_R2.fastq -threads 12 --report /proj/sens2021578/nobackup/emmai/test_bracken/kraken_reports/NM177paired.kreport > /proj/sens2021578/nobackup/emmai/test_bracken/kraken_output/NM177paired.kraken
Loading database information... done.
13716 sequences (6.89 Mbp) processed in 1.351s (609.3 Kseq/m, 305.87 Mbp/m).
  11054 sequences classified (80.59%)
  2662 sequences unclassified (19.41%)

>bracken -d /proj/sens2021578/nobackup/emmai/test_bracken/mydb -i /proj/sens2021578/nobackup/emmai/test_bracken/kraken_reports/NM177paired.kreport -o NM177paired.bracken -r 250
Error: no reads found. Please check your Kraken report

>head NM177paired.kreport 
 19.41  2662    2662    U       0       unclassified
 80.59  11054   50      R       1       root
 46.89  6432    0       D       10239     Viruses
 46.62  6394    0       D1      2731342     Monodnaviria
 46.62  6394    0       K       2732092       Shotokuvirae
 46.62  6394    0       P       2732415         Cossaviricota
 46.62  6394    0       C       2732421           Papovaviricetes
 46.62  6394    0       O       2732533             Zurhausenvirales
 46.62  6394    44      F       151340                Papillomaviridae

**Comment by mziegler12** [comment]
I recognized, if i use a classification level that does not exist in the report,  the error looks like that. Check if you have any taxa on species (S) level in your report. Otherwise use the ```-l``` option to the lowest existing rank. Possible options are ```Options = 'D','P','C','O','F','G','S'```

**Comment by mziegler12** [comment]
Did you try the ```-l``` flag for the highest classification level ```'D'```? I know, i wrote this before, but maybe we can exclude that it has something to do with the classification level.

**Comment by emmaivansson** [comment]
> Did you try the `-l` flag for the highest classification level `'D'`? I know, i wrote this before, but maybe we can exclude that it has something to do with the classification level.

I tried that yes, didn't help unfortunately

**Comment by Gian77** [comment]
Dear @mziegler12,

I just tried and it doesnt' work. Please see the code below. Thanks, G.

```
/mnt/home/benucci/anaconda2/envs/bracken/bin/bracken: illegal option -- -
version of Bracken: Usage: bracken -d MY_DB -i INPUT -o OUTPUT -w OUTREPORT -r READ_LEN -l LEVEL -t THRESHOLD
  MY_DB          location of Kraken database
  INPUT          Kraken REPORT file to use for abundance estimation
  OUTPUT         file name for Bracken default output
  OUTREPORT      New Kraken REPORT output file with Bracken read estimates
  READ_LEN       read length to get all classifications for (default: 100)
  LEVEL          level to estimate abundance at [options: D,P,C,O,F,G,S,S1,etc] (default: S)
  THRESHOLD      number of reads required PRIOR to abundance estimation to perform reestimation (default: 0)
 >> Checking for Valid Options...
 >> Running Bracken 
      >> python src/est_abundance.py -i /mnt/home/benucci/genome-pipe-bac-pb/outputs/12_taxcalssabund_kraken2-brachen/pbio-2432.22840.bc1019_BAK8B_OA--bc1019_BAK8B_OA.ccs/kraken2_report.txt -o /mnt/home/benucci/genome-pipe-bac-pb/outputs/12_taxcalssabund_kraken2-brachen/pbio-2432.22840.bc1019_BAK8B_OA--bc1019_BAK8B_OA.ccs/bracken_report.txt -k /mnt/research/bonito_lab/Benucci/databases/kraken2/minikraken2_v1_8GB/database100mers.kmer_distrib -l D -t 10
>> Checking report file: /mnt/home/benucci/genome-pipe-bac-pb/outputs/12_taxcalssabund_kraken2-brachen/pbio-2432.22840.bc1019_BAK8B_OA--bc1019_BAK8B_OA.ccs/kraken2_report.txt
Error: no reads found. Please check your Kraken report
PROGRAM START TIME: 07-15-2022 18:06:48

```

**Comment by tuncK** [comment]
I ran into the same error, and it looked as if there needs to be at least threshold-many reads in at least one of the user-specified taxonomy levels. I dont know if this helps anyone else. According to the manual, the threshold is 10 by default (which differs from the conda-based post above).

**Comment by tuncK** [comment]
I think if the reason is indeed what I am suspecting, I want to say that the program should actually continue as is but give a column of zeros as output. When it throws such an error, it triggers a cascade of other errors in the pipelines whereas there actually is not much to be checked or fixed.

**Comment by Gian77** [comment]
@tunck

I run it on a genome with 39 conitgs and worked well at `-l G` or `-l S`. I then run it on a genome with 8 contigs and gave me the same error I got ont he genome with 2 contigs. 
I get it but I still do not understand why that threashold of 10 matters for the tool, though...
G.
